src/components/MAPI/releases/permissions/usePermissionsForReleases.ts

```diff
@@ -1,58 +1,82 @@
-import { usePermissions } from 'MAPI/permissions/usePermissions';
-import { hasPermission } from 'MAPI/permissions/utils';
+import { useAuthProvider } from 'Auth/MAPI/MapiAuthProvider';
+import {
+  fetchAccessForResource,
+  fetchAccessForResourceKey,
+} from 'MAPI/permissions/utils';
+import { extractErrorMessage } from 'MAPI/utils';
+import { GenericResponseError } from 'model/clients/GenericResponseError';
 import { Providers } from 'model/constants';
+import { useEffect } from 'react';
+import useSWR from 'swr';
+import { FlashMessage, messageTTL, messageType } from 'utils/flashMessage';
+import { useHttpClientFactory } from 'utils/hooks/useHttpClientFactory';
+
+// eslint-disable-next-line no-magic-numbers
+const REFRESH_INTERVAL = 60 * 1000;
 
 export function usePermissionsForReleases(
   _provider: PropertiesOf<typeof Providers>,
-  namespace: string
+  _namespace: string
```

Comment:
Refactored permissions hook for releases - releases are not a namespace-scoped resource, so it's not technically correct to check if the user has permissions in the `default` namespace. Instead, we perform a `SelfSubjectAccessReview` with namespace argument `''`, meaning all namespaces.